File: files/200-299/253/CH-253.py
import pandas as pd

path = "200-299/253/CH-253 Custom Grouping.xlsx"

input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=25)
test = pd.read_excel(path, usecols="G:H", skiprows=1, nrows=4)

result = input.sort_values(by="Date").reset_index(drop=True)
result['Group'] = (result['Date'].diff().dt.days > 1).cumsum() + 1
result = result.groupby('Group', as_index=False).agg({'Sales': 'sum'})
result = result.rename(columns={'Sales': 'Total Sales'})

print(result.equals(test))

File: files/200-299/253/CH-253.R
library(tidyverse)
library(readxl)

path = "files/200-299/253/CH-253 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:C26")
test  = read_excel(path, range = "G2:H6")

result = input %>%
  arrange(Date) %>%
  mutate(Group = cumsum(c(T, diff(Date)/68400 >= 2))) %>%
  summarise(`Total Sales` = sum(Sales, na.rm = T), .by = Group) 

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE